sdk/core/azure-core/src/main/java/com/azure/core/models/spatial/CollectionGeometry.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models.spatial;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a heterogeneous collection of {@link Geometry geometries}.
+ */
+public final class CollectionGeometry extends Geometry {
+    private final List<Geometry> geometries;
+
+    /**
+     * Constructs a geometry collection.
+     *
+     * @param geometries The geometries in the collection.
+     * @throws NullPointerException If {@code geometries} is {@code null}.
+     */
+    public CollectionGeometry(List<Geometry> geometries) {
+        this(geometries, null, null);
+    }
+
+    /**
+     * Constructs a geometry collection.
+     *
+     * @param geometries The geometries in the collection.
+     * @param boundingBox Bounding box for the geometry collection.
+     * @param properties Additional properties of the geometry collection.
+     * @throws NullPointerException If {@code geometries} is {@code null}.
+     */
+    public CollectionGeometry(List<Geometry> geometries, GeometryBoundingBox boundingBox,
+        Map<String, Object> properties) {
+        super(boundingBox, properties);
+
+        Objects.requireNonNull(geometries, "'geometries' cannot be null.");
+        this.geometries = Collections.unmodifiableList(new ArrayList<>(geometries));
```

Comment:
You shouldn't need the `new ArrayList<>(..)` here, geometries is already a List

Reply:
This is done as the `new ArrayList` will copy the geometry references into an internal representation that the caller won't have a reference to, the latter is to make the representation that may be read unmodifiable.